pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -431,10 +436,19 @@ func (p *Provider) Create(
 		if opts.Lifetime > time.Hour*24 {
 			return errors.New("lifetime cannot be longer than 24 hours for preemptible instances")
 		}
+		if !providerOpts.TerminateOnMigration {
+			return errors.New("preemptible instances require 'TERMINATE' maintenance policy; use --gce-terminateOnMigration")
+		}
 		args = append(args, "--preemptible")
 		// Preemptible instances require the following arguments set explicitly
-		args = append(args, "--maintenance-policy=terminate")
+		args = append(args, "--maintenance-policy", "TERMINATE")
```

Comment:
(I know it's not related to this PR.. but) I don't think we need to set --maintenance-policy and --no-restart-on-faliure when creating a --preemptible vm. though it doesn't matter at all so please feel free to ignore.

https://cloud.google.com/compute/docs/instances/create-use-preemptible#gcloud

Reply:
Right, I was just following the above comment which makes you believe both flags are mandatory :) Either way, I don't see the harm of being explicit.